dfvfs/helpers/data_slice.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""A data slice interface for file-like objects."""
+
+from __future__ import unicode_literals
+
+import os
+
+from dfvfs.lib import py2to3
+
+# Since this class implements the __getitem__ interface the names of
+# the interface functions are in lower case as an exception to the normal
+# naming convention.
+
+
+class DataSlice(object):
+  """Data slice interface for file-like objects."""
+
+  # The maximum allowed size of the read buffer.
+  _MAXIMUM_READ_BUFFER_SIZE = 16 * 1024 * 1024
+
+  def __init__(self, file_object):
+    """Initializes the data slice.
+
+    Args:
+      file_object (FileIO): a file-like object to read from.
+    """
+    super(DataSlice, self).__init__()
+    self._buffer = b''
+    self._file_object = file_object
+    self._file_object_size = file_object.get_size()
+
+  def __enter__(self):
+    """Enters a with statement."""
+    return self
+
+  def __exit__(self, unused_type, unused_value, unused_traceback):
+    """Exits a with statement."""
+    # TODO: do we want to close the file_object here e.g. i.c.w. a flag value
```

Comment:
Please rewrite "e.g. i.c.w". Also, I don't think we want to close the file object here.

Reply:
ack, I've removed the comment.